python/lsst/ts/Dome/mock_controller.py

```diff
@@ -143,120 +211,238 @@ async def cmd_loop(self, reader, writer):
+            string value (CW or CCW) and the key "azRate" with a float value.
+        """
+        self.log.debug(f"Received command 'crawlAz' with arguments {kwargs}")
+
+        # No conversion from radians to degrees needed since both the commands and the mock az controller
+        # use degrees.
+        await self.amcs.crawlAz(
+            direction=kwargs["dirMotion"], velocity=float(kwargs["azRate"])
+        )
+
+    async def crawlEl(self, **kwargs):
+        """Mock crawling the light and wind screen.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "dirMotion" with a
+            string value (UP or DOWN) and the key "elRate" with a float value.
+        """
+        self.log.info(f"Received command 'crawlEl' with arguments {kwargs}")
+        await self.lwscs.crawlEl(
+            direction=kwargs["dirMotion"], velocity=float(kwargs["elRate"])
+        )
+
+    async def setLouver(self, **kwargs):
+        """Mock setting the position of a louver.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain the key "id" with an int
+            value and the key "position" with a float value.
+        """
+        self.log.info(f"Received command 'setLouver' with arguments {kwargs}")
+        await self.lcs.setLouver(
+            louver_id=int(kwargs["id"]), position=int(kwargs["position"])
+        )
+
+    async def closeLouvers(self):
+        """Mock closing all louvers.
+        """
+        self.log.info(f"Received command 'closeLouvers'")
+        await self.lcs.closeLouvers()
+
+    async def stopLouvers(self):
+        """Mock stopping the motion of all louvers.
+        """
+        self.log.info(f"Received command 'stopLouvers'")
+        await self.lcs.stopLouvers()
+
+    async def openShutter(self):
+        """Mock opening the shutter.
+        """
+        self.log.info(f"Received command 'openShutter'")
+        await self.apscs.openShutter()
+
+    async def closeShutter(self):
+        """Mock closing the shutter.
+        """
+        self.log.info(f"Received command 'closeShutter'")
+        await self.apscs.closeShutter()
+
+    async def stopShutter(self):
+        """Mock stopping the motion of the shutter.
+        """
+        self.log.info(f"Received command 'stopShutter'")
+        await self.apscs.stopShutter()
 
     async def config(self, **kwargs):
+        """Mock configure the lower level components.
+
+        Parameters
+        ----------
+        kwargs: `dict`
+            A dictionary with arguments to the function call. It should contain keys for all lower level
+            components to be configured with values that are dicts with keys for all the parameters that
+            need to be configured. The structure is
+            "AMCS":
```

Comment:
This is another case where your structure will be lost when the docs are rendered, unless you do something, such as write "The structure is::" (two colons), then a blank line, then the structure indented.

Reply:
Done